BlockServer/runcontrol/runcontrol_manager.py

```diff
@@ -279,11 +279,10 @@ def _get_latest_ioc_start(self):
         """
         latest_ioc_start = self._channel_access.caget(self._prefix
                                                       + RC_START_PV)
-
-        if latest_ioc_start is not None and latest_ioc_start != '':
+        try:
             frmt = '%m/%d/%Y %H:%M:%S'
             latest_ioc_start = datetime.strptime(latest_ioc_start, frmt)
-        else:
+        except:
```

Comment:
We should try and avoid catch-all exceptions. I actually think the `if` here is ok except for two things:

1. In Python we don't need to check strings for None or empty, just do `if [variable_name]`
1. The great sin of the original code here is switching the variable from being a string to the datetime object within the loop. That's why it gets in a muddle. If we just rename the variable assigned from `self._channel_access.caget(self._prefix + RC_START_PV)` it should be fine.